src/golang/lib/engine/aq_engine.go

```diff
@@ -760,22 +781,25 @@ func (eng *aqEngine) execute(
 			if execState.Status == shared.PendingExecutionStatus {
 				err = op.Launch(ctx)
 				if err != nil {
-					return errors.Wrapf(err, "Unable to schedule operator %s.", op.Name())
+					err = errors.Wrapf(err, "Unable to schedule operator %s.", op.Name())
+					return err
 				}
 				continue
 			} else if execState.Status == shared.RunningExecutionStatus {
 				continue
 			}
 
 			if !execState.Terminated() {
-				return errors.Newf("Internal error: the operator is expected to have terminated, but instead has status %s", execState.Status)
+				err = errors.Newf("Internal error: the operator is expected to have terminated, but instead has status %s", execState.Status)
+				return err
```

Comment:
This pattern is fairly unconventional and I worry that we'll forget to set the error before returning in the future. I'm pretty sure what you can do is name the output argument ala:
```
func (eng *aqEngine) execute(
	ctx context.Context,
	workflowDag dag_utils.WorkflowDag,
	workflowRunMetadata *workflowRunMetadata,
	timeConfig *AqueductTimeConfig,
	opExecMode operator.ExecutionMode,
) (err error) {
```

Then you can just `return errors.Newf(...)` directly and that error variable will be set in the defer(). Might be good too double check this but I'm 99% sure I've done this before.

Reply:
will this variable always override the current `err` variable set in the function body?

Reply:
It should be equivalent to a `var err error` defined at the very beginning of the body. Golang should complain if you attempt to do any inappropriate local context overwrites I think?